content/sensu-enterprise/3.1/guides/using-ec2.md

```diff
@@ -0,0 +1,253 @@
+    - [Integration Configuration](#integration-configuration)
+    - [Client Configuration](#client-configuration)
+  - [IAM Roles and Sensu](#iam-roles-and-sensu)
+  - [Cross-Account IAM Roles](#cross-account-iam-roles)
+  - [Wrapping Up](#wrapping-up)
+  - [References](#references)
+
+# Prerequisites
+
+Before diving into this guide, we recommend having the following components ready:
+
+- A working Sensu Enterprise deployment
+- An AWS account
+- An AWS access key and secret key
+
+If you've not already signed up for Sensu Enterprise, you can do so via [this link][1]. The same goes for an AWS account. You can sign up for an account [here][2]. Once you've signed up for an AWS account, you'll also need to provision an access key and secret key for the user you want to use for Sensu Enterprise. AWS provides a [useful guide for provisioning those keys][3] that you can use to get started.
+
+_NOTE: Make sure you save the CSV with your AWS access and secret key in a safe location._
+
+## EC2 Integration Basics
+
+Sensu Enterprise comes with a built-in integration for handling automatic deregistration of AWS clients. This is especially useful in environments where infrastructure is ephemeral. We'll start off by configuring the integration using the access key and secret key we provisioned for our AWS user.
+
+### Integration Configuration
+
+Our integration configuration file for EC2 will live at `/etc/sensu/conf.d/ec2.json`. Let's take a look at an example:
+
+{{< highlight json >}}
+{
+  "ec2": {
+    "region": "us-west-2",
+    "access_key_id": "AlygD0X6Z4Xr2m3gl70J",
+    "secret_access_key": "y9Jt5OqNOqdy5NCFjhcUsHMb6YqSbReLAJsy4d6obSZIWySv",
+    "allowed_instance_states": ["running"],
+    "timeout": 10
+  }
+}{{< /highlight >}}
+
+When using the EC2 integration in this fashion, the only two required attributes are `access_key_id` and `secret_access_key`. The rest of the attributes are optional, but let's take a look over them
+
+`region`: The `region` attribute is the region that the integration will use for queries about the instance's state. By default, this is set to `us-east-1` (the "Virginia" region).
+
+`allowed_instance_states`: The `allowed_instance_states` attribute determines the state that an instance can be in before it is automatically deregistered. By default, this is set to "running", but you can also set it to "running", "stopping", "stopped", "shutting-down", "terminated", "rebooting", and "pending". Any instance that is in a state not specified in the configuration will be deregistered.
+
+You can use the above example configuration and substitute your own access key and secret key in for the ones in the example. Let's continue and take a look at what is necessary at the client level for the EC2 integration to function properly.
+
+_NOTE: Don't forget to restart the Sensu Enterprise process via `systemctl restart sensu-enterprise` for the configuration changes to take effect._
+
+### Client Configuration
+
```

Comment:
For cross-account usage, the client ec2 attributes need to include the account name, e.g. `{ "client": { "ec2": { "account": "sensu-testing" } } }`